### DISCUSSION TITLE
Multiple config files in Cypress 10?

### DISCUSSION THREAD

**Discussion by shutchison-bai** [cypress-io/cypress#22517] [Questions and Help] [answered]
I currently run my e2e tests on multiple environments.  To achieve this I have created a config file for each environment and pass the configFile value in the CLI.  When upgrading I realized Cypress 10.0.0  does not support multiple config files.  Is there away to achieve this in the latest version of Cypress?

**Comment by phaterster** [comment]
Why it is not supported? It is supported! have you tried?
https://docs.cypress.io/api/plugins/configuration-api#Switch-between-multiple-configuration-files

**Reply by Victorgabarullo** [reply]
I got the same problem! When I clicked on that link it says that is no longer supported

**Comment by stavares843** [comment]
hello! 👋 

have you tried `[these](https://docs.cypress.io/guides/references/migration-guide#Configuration-File-Changes)` instructions for Cypress 10 regarding the config files?

**Comment by bennymeade** [comment, marked as answer]
This does work in version 10 and you can extend from the default config:

I put a mobile config file under ```cypress/configs/cypress.mobile.config.ts```

Then reference it in `package.json` scripts:
```"cy:open:mobile": "cypress open --e2e --browser chrome --config-file cypress/configs/cypress.mobile.config.ts",```

File `cypress.mobile.config.ts` contents:
```
import { defineConfig } from 'cypress';
import defu from 'defu';

import defaultConfig from '../../cypress.config';

export default defineConfig(
  defu(
    {
      viewportWidth: 375,
      viewportHeight: 812,
      userAgent: 'Mozilla/5.0 (iPhone; CPU iPhone OS 13_2_3 like Mac OS X) Mobile/15E148',
      env: {
        device: 'mobile',
      },
      e2e: {
        excludeSpecPattern: ['**/e2e/site-monitoring/*', '**/e2e/keyboard-navigation/*'],
      },
    },
    defaultConfig
  )
);
```

**Reply by notfounnd** [reply]
Hi @bennymeade. Would you have any repo with this example?

**Reply by stavares843** [reply]
we have various Cypress config's on our [repo](https://github.com/Satellite-im/Core-PWA) 🔨

**Reply by notfounnd** [reply]
@stavares843 I thought about this way in a first moment.

But it worry me 'cause in this way I'll duplicate commons configurations of project.

E.g.: Reports configuration (mochawesome), timeouts, screen size, assets folders...

This configurations are the same for all environments, it not seens a good practice duplicate this configs in each file xyz.config.js.

**Comment by cristiannzero** [comment]
Simpler way to do this:

1. Export a module with your common attributes.
`module.exports = {
  projectId: "adfadfasdfas",
  project: "./spec",
  chromeWebSecurity: false,
  retries: {
    runMode: 1,
    openMode: 1,
  },
  pageLoadTimeout: 200000,
  responseTimeout: 700000,
  video: false,
  watchForFileChanges: false
}
`
1. Create your different config files and call this module. Use spread javascript call to merge these properties.

`const { defineConfig } = require("cypress")
const defaultConfig = require("../../cypress.default.config")

module.exports = defineConfig({
  ...defaultConfig,
  defaultCommandTimeout: 20000,
  e2e: {
    baseUrl: "your url",
    specPattern: "cypress/e2e/**/*.{js,jsx,ts,tsx}",
  },
})
`
This way you avoid to have duplicates on each config file

**Comment by dominiknapik** [comment]
You might want to check Gleb's blog post about this issue - https://glebbahmutov.com/blog/load-cypress-env-settings/
